python/cudf/cudf/core/multiindex.py

```diff
@@ -1302,6 +1302,54 @@ def _poplevels(self, level):
 
         return popped
 
+    @_cudf_nvtx_annotate
+    def swapij(self, i=0, j=1):
+        """
+        Swap level i with level j.
+        Calling this method does not change the ordering of the values.
+
+        Parameters
+        ----------
+        i : int, str, default 0
+            First level of index to be swapped.
+        j : int, str, default 1
+            Second level of index to be swapped.
+
+        Returns
+        -------
+        MultiIndex
+            A new MultiIndex.
+
+
+        Examples
+        --------
+        >>> import cudf
+        >>> mi = cudf.MultiIndex(levels=[['a', 'b'], ['bb', 'aa']],
+        ...                    codes=[[0, 0, 1, 1], [0, 1, 0, 1]])
+        >>> mi
+        MultiIndex([('a', 'bb'),
+            ('a', 'aa'),
+            ('b', 'bb'),
+            ('b', 'aa')],
+           )
+        >>> mi.swapij(0, 1)
+        MultiIndex([('bb', 'a'),
+            ('aa', 'a'),
+            ('bb', 'b'),
+            ('aa', 'b')],
+           )
+        """
+        result = self.copy()
+        new_levels = result.levels
+        new_codes = result.codes
+        new_names = list(result.names)
```

Comment:
This implementation of swapping is unfortunately going to be extremely inefficient. Computing levels and codes for our MultiIndex class is sadly very expensive. I suspect that you will end up with a solution that is possibly orders of magnitude more efficient by instead constructing a dictionary of columns manually and using `MultiIndex._from_data` instead. Here is a sample implementation (not tested, and I suspect that you will need to add a little testing for some edge cases)
```
    def swapij(self, i=0, j=1):
        name_i = self._data.names[i]
        name_j = self._data.names[j]

        new_data = {}
        for k, v in self._data.items():
            if k not in (name_i, name_j):
                new_data[k] = v
            elif k == name_i:
                new_data[name_j] = self._data[name_j]
            elif k == name_j:
                new_data[name_i] = self._data[name_i]

        return MultiIndex._from_data(new_data)
```